### PR TITLE
Creating a uAttributes Builder Class

### PR DESCRIPTION
**Pull request by moniercoding**

### CODE REVIEW

File: include/uprotocol-cpp/rpc/RpcClient.h [hunk 1]
```diff
@@ -27,7 +27,8 @@
 
 #include <future>
 #include <uprotocol-cpp/transport/datamodel/UPayload.h>
-#include <uprotocol-cpp/transport/datamodel/UAttributes.h>
+#include <uprotocol-cpp/transport/datamodel/UAttributesBuilder.h>
```
Comment (MelamudMichael) [inline include/uprotocol-cpp/rpc/RpcClient.h:30]: this include is not needed

File: include/uprotocol-cpp/transport/datamodel/UAttributesBuilder.h [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _UATTRIBUTESBUILDER_
+#define _UATTRIBUTESBUILDER_
+
+#include <memory>
```
Comment (MelamudMichael) [inline include/uprotocol-cpp/transport/datamodel/UAttributesBuilder.h:29]: please check if all of the includes are actually needed

File: include/uprotocol-cpp/transport/datamodel/UAttributesBuilder.h [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+#include <up-core-api/uri.pb.h>
+#include <up-cpp/uuid/factory/UuidFactory.h>
+#include <string>
+#include <spdlog/spdlog.h>
+#include "up-core-api/uattributes.pb.h"
```
Comment (MelamudMichael) [inline include/uprotocol-cpp/transport/datamodel/UAttributesBuilder.h:34]: change to <>

File: include/uprotocol-cpp/transport/datamodel/UAttributesBuilder.h [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+using namespace uprotocol::v1;
+
+namespace uprotocol {
+
+class UAttributesBuilder {
+    private:
+        uprotocol::v1::UAttributes attributes_;
+
+    public:
+        UAttributesBuilder() {
```
Comment (MelamudMichael) [inline include/uprotocol-cpp/transport/datamodel/UAttributesBuilder.h:48]: you need also the constructor for the mandatory attributes 
            UAttributes(const UUID id,
                        const UMessageType type,
                        const UPriority priority) {

File: include/uprotocol-cpp/transport/datamodel/UListener.h [hunk 5]
```diff
@@ -27,10 +27,10 @@
 #define _ULISTENER_H_
 
 #include <uprotocol-cpp/uri/datamodel/UUri.h>
-#include <uprotocol-cpp/transport/datamodel/UAttributes.h>
+#include <uprotocol-cpp/transport/datamodel/UAttributesBuilder.h>
```
Comment (MelamudMichael) [inline include/uprotocol-cpp/transport/datamodel/UListener.h:30]: not needed

**Review by MelamudMichael — CHANGES_REQUESTED** [review]

**Review by MelamudMichael — CHANGES_REQUESTED** [review]
This PR is OK , but it should not be merged before the up-client PR is ready
did you run the unit-tests?

**Comment by stevenhartley** [comment]
Why are we not moving to UAttributes to align with core-api 1.5.6?

**Comment by moniercoding** [comment]
Creating a new PR

**Pull request closed without merging**